DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -1,31 +1,97 @@
-﻿namespace DTML.EduBot.Dialogs
+﻿using Autofac;
+using DTML.EduBot.Extensions;
+using Microsoft.Bot.Builder.Internals.Fibers;
+
+namespace DTML.EduBot.Dialogs
 {
     using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
     using System.Threading.Tasks;
     using Microsoft.Bot.Builder.Dialogs;
-    using Microsoft.Bot.Builder.Luis;
-    using Microsoft.Bot.Builder.Luis.Models;
-    using DTML.EduBot.Common;
-    using DTML.EduBot.Qna;
+    using Microsoft.Bot.Connector;
 
-    [LuisModel("", "")]
-    [QnaModel(hostUri: "", subscriptionKey: "", modelId: "")]
     [Serializable]
-    public partial class RootDialog : QnaLuisDialog<object>
+    public class RootDialog : IDialog<string>
     {
-        [LuisIntent("")]
-        [LuisIntent("None")]
-        public async Task HandleUnrecognizedIntent(IDialogContext context, LuisResult result)
+        private const string ChatWithBot = "Chat With Bot";
+        private const string StartTheLessonPlan = "Start The Lesson Plan";
+
+        private static readonly IEnumerable<string> Choices = new ReadOnlyCollection<string>
+            (new List<String> {
+                ChatWithBot,
+                StartTheLessonPlan});
+
+        public async Task StartAsync(IDialogContext context)
+        {
+            context.Wait(this.MessageReceivedAsync);
+        }
+
+        private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
         {
-            await context.PostAsync(BotPersonality.BotResponseUnrecognizedIntent);
+            string friendlyUserName = context.Activity.From.Name;
+            
+            PromptDialog.Choice(
+                context,
+                this.AfterChoiceSelected,
+                Choices,
+                "Hello Dear " + friendlyUserName + ",\n What would you like to do.",
+                "I am sorry but I didn't understand that. I need you to select one of the options below",
+                attempts: Choices.Count());
         }
 
-       [LuisIntent("Gibberish")]
-        public async Task HandleGibberish(IDialogContext context, LuisResult result)
+        private async Task AfterChoiceSelected(IDialogContext context, IAwaitable<string> result)
         {
-            await context.PostAsync(BotPersonality.BotResponseToGibberish);
+            try
+            {
+                var selection = await result;
+
+                using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+                {
+                    switch (selection)
+                    {
+                        case ChatWithBot:
```

Comment:
I might be missing something, but did you mention that you were using a bool now?